openpathsampling/experimental/storage/test_monkey_patches.py

```diff
@@ -0,0 +1,43 @@
+import openpathsampling as paths
+import pytest
+
+from openpathsampling.experimental.storage.monkey_patches import *
+from openpathsampling.experimental.storage.collective_variables import \
+    CollectiveVariable
+
+@pytest.fixture
+def patching_tps_network():
+    cv = CollectiveVariable(lambda x: x.xyz[0][0]).named('cv')
+    state_A = paths.CVDefinedVolume(cv, 0, 1)
+    state_B = paths.CVDefinedVolume(cv, 9, 10)
+    tps_network = paths.TPSNetwork(state_A, state_B)
+    yield tps_network
+
+
+def test_double_patch(patching_tps_network):
+    orig_dict = patching_tps_network.to_dict()
+    global paths
+    paths = monkey_patch_all(paths)
+    try:
+        patched_dict = patching_tps_network.to_dict()
+        paths = monkey_patch_all(paths)
+        double_patched_dict = patching_tps_network.to_dict()
+    finally:
+        unpatch(paths)
+
+    assert orig_dict != patched_dict
+    assert patched_dict == double_patched_dict
+
+
+def test_unpatch(patching_tps_network):
+    orig_dict = patching_tps_network.to_dict()
+    global paths
+    paths = monkey_patch_all(paths)
+    try:
+        patched_dict = patching_tps_network.to_dict()
+    finally:
+        paths = unpatch(paths)
```

Comment:
same as above
```suggestion
    try:
        patched_dict = patching_tps_network.to_dict()
    except:
        unpatch(paths)
        raise
    else:
        paths = unpatch(paths)
```

Reply:
I think you meant to add the rest of the test to the try part of the suggestion?  Is the suggestion, as is, any different in practice from the current code? I thought that just using `finally` would be the same as `except: do_something(); raise` followed by `else: do_something()`?

Reply:
Sorry, I misremembered how a `try` without an `except` worked. I somehow was convinced it would catch `all` (don't know why I thought that. :facepalm:  )

you code is better than my suggested change.